bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Hors.java

```diff
@@ -0,0 +1,168 @@
+            throw new BankException(
+                    res.getText(R.string.unable_to_find).toString() + " EventValidation.");
+        }
+        String eventValidation = e.attr("value");
+
+        List<NameValuePair> postData = new ArrayList<NameValuePair>();
+        postData.add(new BasicNameValuePair("__EVENTTARGET", ""));
+        postData.add(new BasicNameValuePair("__EVENTARGUMENT", ""));
+        postData.add(new BasicNameValuePair("__VIEWSTATE", viewState));
+        postData.add(new BasicNameValuePair("__VIEWSTATEGENERATOR", "D7823E0B"));
+        postData.add(new BasicNameValuePair("__EVENTVALIDATION", eventValidation));
+        postData.add(new BasicNameValuePair("_ctl0:cphMain:SmartpassTextbox", getUsername()));
+        postData.add(new BasicNameValuePair("_ctl0:cphMain:SubmitButton", "OK"));
+        postData.add(new BasicNameValuePair("_ctl0:cphMain:chbRememberSmartPassCode", "on"));
+        postData.add(new BasicNameValuePair("_ctl0:cphMain:cookieEnabled", "true"));
+        return new LoginPackage(urlopen, postData, response, "https://www.dittkort.se//q/?p=7EB4F129-0A41-417F-8FEA-51B2B75B9D24");
+    }
+
+    public Urllib login() throws LoginException, BankException, IOException {
+        LoginPackage lp = preLogin();
+        response = urlopen.open(lp.getLoginTarget(), lp.getPostData());
+        if (response.contains("cphMain_txtGeneralCardError")) {
+            throw new LoginException(res.getText(R.string.invalid_card_number).toString());
+        }
+        return urlopen;
+    }
+
+    @Override
+    public void update() throws BankException, LoginException, BankChoiceException, IOException {
+        super.update();
+        if (getUsername().isEmpty()) {
+            throw new LoginException(res.getText(R.string.invalid_card_number).toString());
+        }
+        urlopen = login();
+        Document d = Jsoup.parse(response);
+        Element e = d.getElementById("cphMain_lblAmount");
+        if (e == null) {
+            throw new BankException(
+                    res.getText(R.string.unable_to_find).toString() + res.getText(R.string.balance).toString());
+        }
+        BigDecimal b = Helpers.parseBalance(e.text());
+        e = d.getElementById("lblCardName");
+        String name = "HÖRS";
+        if (e != null) {
+            name = e.text();
+            if (this.getCustomName().isEmpty()) {
+                this.setCustomName(name);
+            }
+        }
+        Account a = new Account(name, b, "0");
+        accounts.add(a);
+        balance = balance.add(a.getBalance());
+
+        d = Jsoup.parse(urlopen.open("https://www.dittkort.se/q/Partial/Transactions.aspx?cnt=20"));
+
+        Elements es = d.select("tr");
+        List<Transaction> transactions = new ArrayList<Transaction>();
+        if (es != null) {
+            for (Element el : es) {
+                transactions.add(
+                        new Transaction(el.child(0).text().trim(), el.child(1).text().trim(),
```

Comment:
Maybe break out this line to a private method named `Transaction asTransaction(Element element)`